impeller/compiler/compiler.cc

```diff
@@ -464,13 +464,24 @@ const std::vector<std::string>& Compiler::GetIncludedFileNames() const {
   return included_file_names_;
 }
 
+// Escape `%` and `#` characters according to doc comment at
+// https://github.com/ninja-build/ninja/blob/master/src/depfile_parser.cc#L28
+static void EscapeString(std::string& str, std::stringstream& stream) {
```

Comment:
`CanCompileAndReflect` [here](https://github.com/flutter/engine/blob/main/impeller/compiler/compiler_test.cc#L61) could have some logic added to it to emit and inspect the contents of the depfile.

Reply:
I gave it a shot...